File: javascript/gigasecond/gigasecond.js
var Gigasecond = function(startDate) {
  this.startDate = startDate;
}

Gigasecond.prototype.date = function() {
  var startTime = this.startDate.getTime();
  var endDate = new Date(startTime + Math.pow(10, 12));

  return endDate;
}

module.exports = Gigasecond;

File: java/gigasecond/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;

class Gigasecond {
  LocalDateTime birthDateTime;
  long gigaseconds = (long) Math.pow(10, 9);

    Gigasecond(LocalDate birthDate) {
      this.birthDateTime = birthDate.atStartOfDay();
    }

    Gigasecond(LocalDateTime birthDateTime) {
      this.birthDateTime = birthDateTime;
    }

    LocalDateTime getDate() {
      return this.birthDateTime.plusSeconds(gigaseconds);
    }

}
